cirq/ops/display.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes to represent displays.
+
+A Display is an operation that signifies extracting some information about the
+qubits it is applied to without actually performing any effect on those qubits.
+Each Display in a circuit has an associated key, and the result of computing
+the values of Displays in a circuit is a dictionary from Display key to
+Display value.
+"""
+
+from typing import Any, Dict, Hashable, Tuple, TYPE_CHECKING
+
+import abc
+
+import numpy as np
+
+from cirq import protocols, value
+from cirq.ops import raw_types, op_tree
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from cirq.ops import pauli_string
+
+
+class SamplesDisplay(raw_types.Operation):
+    """A display whose value is computed from measurement results.
+
+    The value of a SamplesDisplay on some qubits is computed in the following
+    steps:
+        1. Perform some unitary operations on the qubits
+        2. Sample bitstrings from the resulting state a number of times
```

Comment:
This description is not correct. It sounds like you're describing repeated measurement of one prepared state, instead of repeated prepare+measure.

Reply:
I've made it hopefully clearer.